### ISSUE TITLE
Google Calendar - htmlLink not always defined

### ISSUE THREAD

**Issue by bensinclair** [fullcalendar/fullcalendar#2844]
Over the weekend I started getting the following error.

![screen shot 2015-08-31 at 11 20 21 am](https://cloud.githubusercontent.com/assets/1043658/9570555/4930c63e-4fd2-11e5-977d-3f0e8fbc628e.png)

It's within the `injectQsComponent` function. It appears not all items in a Google Calendar events array have a `htmlLink` key. 

Here's an example:

````json
  {
   "kind": "calendar#event",
   "etag": "\"2881697643358000\"",
   "id": "xxxxxxxxxx",
   "status": "confirmed",
   "updated": "2015-08-29T11:47:01.679Z",
   "start": {
    "dateTime": "2015-09-03T04:30:00-04:00",
    "timeZone": "Australia/Sydney"
   },
   "end": {
    "dateTime": "2015-09-03T05:15:00-04:00",
    "timeZone": "Australia/Sydney"
   },
   "visibility": "private",
   "iCalUID": "xxxxxxxxxx@google.com"
  }
````

Doesn't happen on all calendars. Would just be a case of checking if `entry.htmlLink` is defined. To fix temporarily this is what I added on line 139:

```javascript
var url = typeof(entry.htmlLink) != 'undefined' ? entry.htmlLink : '';
````

**Comment by bensinclair** [comment]
Just to expand on this, I beleive this is an issue with Google Apps calendars. I think Google must be tightening security. Under Calendar settings in Google Apps, it is set to show "Only free/busy information (hide event details)" which is what I think is causing this. I've switch to to publis show this (which can take 24 hours to update) so I'll let you know the results.

It might just be a privacy setting afterall.

**Comment by espen** [comment]
Are you using the gcal as mentioned here? http://fullcalendar.io/docs/google_calendar/

The demo at http://fullcalendar.io/js/fullcalendar-2.4.0/demos/gcal.html seems to work.

**Comment by bensinclair** [comment]
The demo isn't using a calendar attached to a Google Apps account (see my last comment). I'll let you know if me changing my security settings resolves the issue.

**Comment by bensinclair** [comment]
Please see attached. 

![google apps calendar sharing](https://cloud.githubusercontent.com/assets/1043658/9891500/00a9cba2-5c4a-11e5-9893-07f935e4f21e.png)

If you have not enabled sharing in Google Apps it doesn't provide things like the event name and the link. Obviously this is user error but maybe more graceful erroring would be useful?

**Comment by arshaw** [comment]
i agree, a JS error should be avoided. there should be an existence check for htmlLink

**Comment by arshaw** [comment]
fixed with 7dc98d900875d79920a657e63ea073d4aa32ea74. just released
